### PR TITLE
[BUG] Fix query materialized view bug

### LINKED ISSUE #7914: query materialized view failed
### Steps to reproduce the behavior (Required)
1. CREATE MATERIALIZED VIEW part_mv
distributed by hash(`p_partkey`) BUCKETS 3
REFRESH MANUAL
AS
SELECT p_partkey, p_name, sum(p_size) as total_size, count(p_container) as count_container
FROM part
GROUP BY p_partkey, p_name;
2.
insert into part_mv
SELECT p_partkey, p_name, sum(p_size) as total_size, count(p_container) as count_container
FROM part
GROUP BY p_partkey, p_name;

3. select count(*) from part_mv;

### Expected behavior (Required)
query materialized view successfully
### Real behavior (Required)
query failed. error message as following:
Not support table type: MATERIALIZED_VIEW
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7914 

## Problem Summary(Required) ：

Now query against materialized view is not supported and will lead to error. This pr add support to query materialized view.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java [hunk 1]
```diff
@@ -394,7 +394,8 @@ public LogicalPlan visitTable(TableRelation node, ExpressionMapping context) {
         Map<Column, ColumnRefOperator> columnMetaToColRefMap = columnMetaToColRefMapBuilder.build();
         List<ColumnRefOperator> outputVariables = outputVariablesBuilder.build();
         LogicalScanOperator scanOperator;
-        if (node.getTable().getType().equals(Table.TableType.OLAP)) {
+        if (node.getTable().getType().equals(Table.TableType.OLAP)
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java:397]: After this PR https://github.com/StarRocks/starrocks/pull/7896 is merged, you can use node.getTable().isNativeTable()

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/RelationTransformer.java:397]: ok.

**Comment by ABingHuang** [comment]
fixed by https://github.com/StarRocks/starrocks/pull/7896
close this pr

**Pull request closed without merging**